src/main/java/com/bina/varsim/fastqLiftover/GenomeLocation.java

```diff
@@ -3,13 +3,20 @@
 public class GenomeLocation implements Comparable<GenomeLocation> {
     public String chromosome;
     public int location = 0;
+    public int read_location1 = -1; // if >=0, this is the 1-base index corresponding to a location in a read
```

Comment:
make this a class called "Integer1", then you can use "null" to indicate that it doesn't exist and also capture to information for a 1-based unsigned integer. Eventually we will need to change all the similar variables to capture the type.